influxdb3_write/src/write_buffer/buffer_segment.rs

```diff
@@ -329,17 +325,22 @@ impl DatabaseBuffer {
         table_batch: TableBatch,
     ) {
         if !self.table_buffers.contains_key(&table_name) {
-            // TODO: this check shouldn't be necessary. If the table doesn't exist in the catalog
-            //      and we've gotten here, it means we're dropping a write.
-            if let Some(table) = self.db_schema.get_table(&table_name) {
+            if let Some(table) = self
+                .catalog
+                .db_schema(&self.db_name)
+                .expect("db should exist in schema")
+                .get_table(&table_name)
+            {
                 self.table_buffers.insert(
                     table_name.clone(),
                     TableBuffer::new(segment_key.clone(), &table.index_columns()),
                 );
             } else {
-                return;
+                // Sanity check panic in case this isn't true
+                unreachable!("table should exist in schema");
```

Comment:
It would have the same effect, but in a few less lines, if you `expect` on the `get_table`, e.g.,
```rust
let table = self
    .catalog
    .db_schema(&self.db_name)
    .expect("db should exist in schema")
    .get_table(&table_name)
    .expect("table should exist in schema"); // 👈

self.table_buffers.insert(
    /* ... */
```

Reply:
Unfortunately that's a borrow check error. I tried that way originally and couldn't wrangle it to be better.

Reply:
Aw, dang - okay, I will resolve!